### PR TITLE
scala lightGBM classifier and regressor[WIP]

### PR DESCRIPTION
**Pull request by songhappy**
### 1. Why the change?
to support LightGBM on BigDL

### 2. Summary of the change 
add LightGBMClassifier, LightGBMClasifierModel, LightGBMRegressor, LightGBMRegressorModel
based on  https://github.com/microsoft/SynapseML/tree/master/lightgbm

### 3. How to test?
- [ ] Unit test
 com.intel.analytics.bigdl.dllib.nnframes.LightGBMTrainSpec

### 4. New dependencies
- [ ] New Java/Scala dependencies and their license
       - MIT license, 
        <dependency>
            <groupId>com.microsoft.azure</groupId>
            <artifactId>synapseml-lightgbm_2.12</artifactId>
            <version>0.10.0</version>
        </dependency>

### CODE REVIEW

**Comment by jason-dai** [comment]
1) Move both XGBoost and LightGBM to a different file (e.g., TreeMode)
2) What's the size the of the new dependency?

**Comment by jason-dai** [comment]
Break this PR into two: scala version first, and then python version

File: scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/nnframes/TreeMode.scala [hunk 1]
```diff
@@ -0,0 +1,593 @@
+/*
```
Comment (jason-dai) [inline scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/nnframes/TreeMode.scala:1]: treemodel

File: scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/nnframes/TreeMode.scala [hunk 2]
```diff
@@ -0,0 +1,797 @@
...
+}
+
+object LightGBMClassifierModel {
+  def loadNativeModel(path: String): LightGBMClassifierModel = {
+    new LightGBMClassifierModel(MLightGBMClassificationModel.loadNativeModelFromFile(path))
+  }
+}
+
+/**
+ * [[XGBRegressor]] xgboost wrapper of XGBRegressor.
```
Comment (dding3) [inline scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/nnframes/TreeMode.scala:611]: LightGBMRegressor?

File: scala/dllib/src/test/scala/com/intel/analytics/bigdl/dllib/nnframes/LightGBMTrainSpec.scala [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+import com.microsoft.azure.synapse.ml.lightgbm.{LightGBMClassifier => MLightGBMClassifier}
+import org.apache.spark.SparkConf
+
+
+class LightGBMTrainSpec extends ZooSpecHelper {
+  var sc : SparkContext = _
+  var sqlContext : SQLContext = _
+
+  override def doBefore(): Unit = {
+    val conf = new SparkConf().setAppName("Test NNClassifier").setMaster("local[1]")
```
Comment (dding3) [inline scala/dllib/src/test/scala/com/intel/analytics/bigdl/dllib/nnframes/LightGBMTrainSpec.scala:33]: Test lightGBM?

File: scala/dllib/src/test/scala/com/intel/analytics/bigdl/dllib/nnframes/XgboostTrainSpec.scala [hunk 4]
```diff
@@ -90,9 +86,8 @@ class XgboostTrainSpec extends ZooSpecHelper {
       xgbRegressorModel0.save("/tmp/test")
       val model = XGBRegressorModel.load("/tmp/test")
       val y0_0 = model.transform(assembledDf)
-      TestUtils.conditionFailTest(y0_0.except(y0).count()==0)
+      y0_0.show()
```
Comment (dding3) [inline scala/dllib/src/test/scala/com/intel/analytics/bigdl/dllib/nnframes/XgboostTrainSpec.scala:89]: why this change here?

File: scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/nnframes/TreeMode.scala (hunk 2 above)
Comment (songhappy) [reply scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/nnframes/TreeMode.scala]: updated

File: scala/dllib/src/test/scala/com/intel/analytics/bigdl/dllib/nnframes/LightGBMTrainSpec.scala (hunk 3 above)
Comment (songhappy) [reply scala/dllib/src/test/scala/com/intel/analytics/bigdl/dllib/nnframes/LightGBMTrainSpec.scala]: updated

File: scala/dllib/src/test/scala/com/intel/analytics/bigdl/dllib/nnframes/XgboostTrainSpec.scala (hunk 4 above)
Comment (songhappy) [reply scala/dllib/src/test/scala/com/intel/analytics/bigdl/dllib/nnframes/XgboostTrainSpec.scala:89]: added assert back

**Comment by songhappy** [comment]
PR-validataion 
http://10.112.231.51:18888/job/BigDL-PRVN-Spark-2.4/629/
failure is irrelevant

**Comment by jason-dai** [comment]
It should be TreeModel instead of TreeMode.

**Comment by dding3** [comment]
LGTM
